src/SFA.DAS.AANHub.Application/Attendances/Commands/PutAttendance/PutAttendanceCommandHandler.cs

```diff
@@ -79,8 +90,47 @@ private async Task<ValidatedResponse<SuccessCommandResult>> CreateAttendance(Put
 
         _attendancesWriteRepository.Create(newAttendance);
 
+        await CreateNotification(command);
+
         await _aanDataContext.SaveChangesAsync(token);
 
         return new ValidatedResponse<SuccessCommandResult>(new SuccessCommandResult());
     }
+
+    private async Task CreateNotification(PutAttendanceCommand command)
+    {
+        var member = await _membersReadRepository.GetMember(command.RequestedByMemberId);
+        var templateName = GetTemplateName(command, member!);
+        var tokens = await GetTokens(command, member!);
+
+        Notification notification = NotificationHelper.CreateNotification(command.RequestedByMemberId, templateName, tokens, true);
+        _notificationsWriteRepository.Create(notification);
+    }
+
+    private static string GetTemplateName(PutAttendanceCommand command, Member member)
+    {
+        var templateName = string.Empty;
+
+        switch (member!.UserType)
+        {
```

Comment:
Also check out pattern matching which would not only simplify this code a lot but also make it easy to read